[DOCS] FDS Topic Quality Review (#7944)
great-expectations/great_expectations: docs/docusaurus/docs/guides/expectations/how_to_create_and_edit_expectations_with_a_profiler.md

    }
```

> The intention is that you'll edit and update the Expectation Suite to better suit your specific use case. The Expectation Suite is not intended to be used without changes.

Reviewer: This information also pertains to the Onboarding Data Assistant, and we may want to include it in the previous document about the ODA, as the User Configurable Profiler is no longer the recommended way to profile data (and so this guide may be removed in the future).

Author: @Rachel-Reverie I've implemented the recommended edit.

```python name="tests/integration/docusaurus/expectations/how_to_create_and_edit_expectations_with_a_profiler create_asset"
```

If a <TechnicalTag tag="datasource" text="Datasource" /> and data asset already exist, run the following command to load an on-disk <TechnicalTag tag="data_context" text="Data Context" />:

```python name="tests/integration/docusaurus/expectations/how_to_create_and_edit_expectations_with_a_profiler get_asset"
```

## 2. Set your expectation_suite_name and create your Batch Request

Run the following command to identify the <TechnicalTag tag="batch" text="Batch" /> of data you want to <TechnicalTag tag="profiling" text="Profile" /> to create your Expectation Suite:

```python name="tests/integration/docusaurus/expectations/how_to_create_and_edit_expectations_with_a_profiler name_suite"
```

## 3. Instantiate your Validator

Run the following command to use a <TechnicalTag tag="validator" text="Validator" /> to access and interact with your data:

```python name="tests/integration/docusaurus/expectations/how_to_create_and_edit_expectations_with_a_profiler create_validator"
```

> Run `validator.head()` to confirm that it contains the expected data.

Reviewer: I'm slightly conflicted.  It's such a short command that having it inline is perfectly sensible, but I'm a strong advocate for having all code we ask someone to run be inside of a code block so we can have it pulled from an example script that is under test.  We can probably do this one ourselves easily enough -- adding `validator.head()` to the existing `tests/integration/docusaurus/expectations/how_to_create_and_edit_expectations_with_a_profiler.py` script is a simple edit.  Ping me if you want to pair and I'll walk you through how our named snippet tags work.

Author: @Rachel-Reverie I have added the edit as requested.

## 4. Instantiate a UserConfigurableProfiler

Run the following command to instantiate a UserConfigurableProfiler and pass in the Validator with your data:

```python name="tests/integration/docusaurus/expectations/how_to_create_and_edit_expectations_with_a_profiler create_profiler"
```

## 5. Use the profiler to build a suite

Run the following command to print a list of all the Expectations created by column, and return the Expectation Suite object:

```python name="tests/integration/docusaurus/expectations/how_to_create_and_edit_expectations_with_a_profiler build_suite"
```

## 6. Running validation, saving your suite, and building Data Docs (Optional)

Run the following command to <TechnicalTag tag="validation" text="Validate" /> your data with the new Expectation Suite, save your Expectation Suite, and build the <TechnicalTag tag="data_docs" text="Data Docs" />:

```python name="tests/integration/docusaurus/expectations/how_to_create_and_edit_expectations_with_a_profiler e2e"
```

## Optional Parameters

> The following are the optional parameters for UserConfigurableProfiler:

Reviewer: I feel like this is something we could handle by linking to the API doc for the `UserConfigurableProfiler`, which would ensure completeness and automatic updates if the list changes in the future (since the docstring that generates the API doc page would be updated).

Side note: generally speaking, any proper class names or parameter names should be wrapped in back ticks.  So: `UserConfigurableProfiler` vs User Configurable Profiler.

Author: @Rachel-Reverie as recommended, I removed the text and bulleted list and linked to the API reference.

- **value_set_threshold**: str: Specify a value from the following ordered list - "none", "one", "two", "very_few", "few", "many", "very_many", "unique". When the Profiler runs, each column is profiled for cardinality. This threshold determines the greatest cardinality to add  `expect_column_values_to_be_in_set`. For example, if `value_set_threshold` is set to "unique", it adds a value_set Expectation for every included column. If set to "few", it adds a value_set expectation for columns whose cardinality is one of "one", "two", "very_few" or "few". The default value here is "many". To compare whether two tables are identical, set the value to "unique".
- **semantic_types_dict**: Dict\[str, List\[str\]\]. See the following section.

To make use of these parameters, you can specify them when you instantiate your Profiler. For example:

```python name="tests/integration/docusaurus/expectations/how_to_create_and_edit_expectations_with_a_profiler optional_params"
```

> **After instantiating a Profiler with specific parameters, you must re-instantiate the Profiler to apply the changes.**

Reviewer: ```suggestion
**To change the parameters of an instantiated Profiler you must re-instantiate the Profiler with the new parameters.**
```

## Semantic Types Dictionary Configuration

> If the Profiler detects that a column is numeric, it creates numeric Expectations. If you don't want numeric Expectations, you can use `semantic_types_dict` to define the Expectations type. The semantic types that you can specify in the UserConfigurableProfiler are "numeric", "value_set", and "datetime". You can pass a dictionary where the keys are the semantic types, and the values are lists of columns of those semantic types.

Reviewer: ```suggestion
If the Profiler detects that a column is numeric, it creates numeric Expectations. If you don't want numeric Expectations, you can use `semantic_types_dict` to define the Expectations type. The semantic types that you can specify in the `UserConfigurableProfiler` are "numeric", "value_set", and "datetime". You can pass a dictionary where the keys are the semantic types, and the values are lists of columns of those semantic types.
```
